pkg/provider/api.go

```diff
@@ -376,6 +399,38 @@ func (ap *ApiProvider) RefreshUsers(ctx context.Context) error {
 		optionLimit  = slack.GetUsersOptionLimit(1000)
 	)
 
+	// Get team ID for cache partitioning
+	authResp, err := ap.client.AuthTest()
+	if err != nil {
+		ap.logger.Error("Failed to get auth test for team ID", zap.Error(err))
+		return err
+	}
+	teamID := authResp.TeamID
+	if teamID == "" {
+		ap.logger.Warn("Team ID is empty, skipping Redis cache operations")
```

Comment:
nit can we move this log to before the return? https://github.com/natomalabs/slack-mcp-server/blob/482d6f5fbaadc80e50dbd18a77554285ddc1a792/pkg/provider/api.go#L317-L319